### PR TITLE
Kernel: Clean up i686 support

### PR DESCRIPTION
**Pull request by boricj**
With these changes, SerenityOS can boot on a canonical i686 CPU, which is defined to be a Pentium Pro processor or better. The original Pentium CPU doesn't support CMOV nor PAE and SerenityOS will therefore not work on it as-is.

Sadly, the museum collection in my basement does not include a Pentium Pro artifact, so I can't test this with period-correct physical hardware... That being said, experiments with QEMU suggests that it should work in theory.

### CODE REVIEW

**Comment by supercomputer7** [comment]
I'm not against supporting older hardware, but do we actually have a way to test this on bare metal ?
i686 (Pentium Pro) was launched in 1995, so I hardly believe anyone will be able to see the difference.

**Comment by boricj** [comment]
It's not (just) about running SerenityOS on old physical hardware. It's about cleaning up some incorrect assumptions about non-SSE x86 CPUs in the kernel, which has code to handle it, as well as explicitly setting the baseline to i686 for virtual machines.

After **much** shuffling around I simply can't find in my basement a "barebones" i686-class processor, which I define as having support for CMOV but not FXSR. However, I did find i686-class processors with CMOV and FXSR but not SSE, which at least would verify that SerenityOS works on non-SSE physical processors.

Sadly, AMD K6-III and Pentium Pro CPUs/computers seems fairly rare and expensive nowadays. I'm not going to buy $100+ parts overseas just to prove the non-FXSR stuff on real hardware.

As an aside, I do have regular Pentium, 80486 and 80386 CPUs, but I suppose I'll need to fork SerenityOS into NostalgiaOS if I was to mess with those relics.

File: Kernel/Arch/i386/CPU.cpp [hunk 1]
```diff
@@ -1412,12 +1420,18 @@ extern "C" void enter_thread_context(Thread* from_thread, Thread* to_thread)
...
     auto& to_tss = to_thread->tss();
-    asm volatile("fxsave %0"
-                 : "=m"(from_thread->fpu_state()));
+
+    if (has_fxsr)
+        asm volatile("fxsave %0"
+                     : "=m"(from_thread->fpu_state()));
+    else
+        asm volatile("fnsave %0"
+                     : "=m"(from_thread->fpu_state()));
```
Comment (awesomekling) [inline Kernel/Arch/i386/CPU.cpp:1435]: Same here.

File: Kernel/Arch/i386/CPU.cpp [hunk 2]
```diff
@@ -1233,8 +1237,12 @@ UNMAP_AFTER_INIT void Processor::initialize(u32 cpu)
...
         VERIFY((FlatPtr(&s_clean_fpu_state) & 0xF) == 0);
         asm volatile("fninit");
-        asm volatile("fxsave %0"
-                     : "=m"(s_clean_fpu_state));
+        if (has_feature(CPUFeature::FXSR))
+            asm volatile("fxsave %0"
+                         : "=m"(s_clean_fpu_state));
+        else
+            asm volatile("fnsave %0"
+                         : "=m"(s_clean_fpu_state));
```
Comment (awesomekling) [inline Kernel/Arch/i386/CPU.cpp:1246]: Please wrap the `if` and `else` blocks in `{ }` since they span multiple lines. :)

**Review by awesomekling — CHANGES_REQUESTED** [review]

File: Kernel/Arch/i386/CPU.cpp (hunk 2 above)
Comment (boricj) [reply Kernel/Arch/i386/CPU.cpp:1246]: The linter is not happy about that. I guess it doesn't want braces on what is technically a single statement.

Comment (awesomekling) [reply Kernel/Arch/i386/CPU.cpp:1246]: That's weird? `clang-format` is (somewhat unfortunately) not opinionated about braces AFAIK.

But okay fine.

**Pull request merged**